app/[team]/home/page.tsx

```diff
@@ -1,16 +1,45 @@
 // all 카테고리 페이지
+//여기서 동적 태그 작업
 import React from 'react';
+import { Metadata, ResolvingMetadata } from 'next';
 
+import { getMetaBlogInfo } from '@/api/blog';
 import { getBlogArticleList, getBlogMainImg } from '@/api/blogHome';
 import ArticleContainer from '@/components/blog/ui/ArticleContainer';
 
+type Props = {
+  params: { team: string };
+  searchParams: { [key: string]: string | string[] | undefined };
+};
+
+// 메타데이터 끼면 에러생김 --> 공문 보고 다시 해보기
+export async function generateMetadata({ params, searchParams }: Props, parent: ResolvingMetadata): Promise<Metadata> {
+  const team = params.team;
+  // const product = await getBlogInfo(team);
+  const product = await getMetaBlogInfo(team);
+  const blogUrl = product.data.blogUrl;
+
+  return {
+    title: product.data.metaName,
+    description: product.data.metaDescription,
+    openGraph: {
+      title: product.data.metaName,
+      description: product.data.metaDescription,
+      type: 'website',
+      url: `${blogUrl}.com`,
+    },
+  };
+}
```

Comment:
```suggestion
if (!product) return null;

const { data: { metaName: title, metaDescription: description } } = product;

  return {
    title,
    description,
    openGraph: {
      title,
      description,
      type: 'website',
      url: `${blogUrl}.com`,
    },
  };
```

요로코롬~유효성 검증이랑 구조분해 할당 써서 코드 간단하게 줄여볼 수도 있을 것 가타우~~

Reply:
You 천재?